kong/runloop/mesh.lua

```diff
@@ -33,8 +34,8 @@ local function nginx_mesh_alpn_select(ssl, protos, mesh_server_ssl_ctx, mesh_alp
       -- https://github.com/openssl/openssl/issues/1652#issuecomment-384660673
       ssl:setVerify(mesh_server_ssl_ctx:getVerify()) -- to set e.g. VERIFY_FAIL_IF_NO_PEER_CERT
       return v
-    -- elseif v == "h2" -- TODO: figure out if current proxy listener directive has http2 allowed
-    elseif v == "http/1.1" then
+    elseif v == "http/1.1" or (v == "h2" and http2_enabled) then
+      -- TODO: figure out if current proxy listener directive has http2 allowed
```

Comment:
@james-callahan as this runs on `init` there is no way to check `current`, or can you elaborate what you mean about `current`?

Reply:
This doesn't run on init, it is called for each incoming connection (this is a callback set to be run for each TLS negotiation)
The question is how do we map from e.g. an `ssl` object to the related proxy listener.

Reply:
oh...  but I tried it with http 1.1 and http 2 and it seems to work still.

Reply:
what happens if we just do `v == "http/1.1" or v == "h2"` here without the `http2_enabled` check?

Reply:
(asking as I would assume that if `http2` is not enabled then it would probably fail somewhere else and letting both pass here would be harmless... I guess?)

Reply:
I was explained out-of-band that this would result in degraded behavior